### PR TITLE
Site config dict constructor

### LINKED ISSUE #555: Python comparison for site_config

### LINKED ISSUE #556: Python constructor for site_config

### PR DESCRIPTION
**Pull request by ManInFez**
requires #561 

resolves #555 
resolves #556

### CODE REVIEW

File: python/res/enkf/site_config.py [hunk 1]
```diff
@@ -13,48 +13,96 @@
...
-    _get_location           = ResPrototype("char* site_config_get_location(site_config)")
+    _get_location           = ResPrototype("char* site_config_get_location()", bind=False)
     _get_config_file        = ResPrototype("char* site_config_get_config_file(site_config)")
     _get_umask              = ResPrototype("int site_config_get_umask(site_config)")
 
 
-    def __init__(self, user_config_file=None, config_content=None):
+    def __init__(self, user_config_file=None, config_content=None, config_dict=None):
 
+        configs = sum([1 for x in [user_config_file, config_content, config_dict] if x is not None])
```
Comment (oysteoh) [inline python/res/enkf/site_config.py:40]: kind of "ugly"...? prettier?
```
configs = [user_config_file, config_content, config_dict].count(None)
```

File: python/res/enkf/site_config.py [hunk 2]
```diff
@@ -13,48 +13,96 @@
...
+                    continue
+                files = os.listdir(job_path)
+                for file_name in files:
+                    full_path = os.path.join(job_path, file_name)
+                    try:
+                        new_job = ExtJob(config_file=full_path, private=False, license_root_path=__license_root_path)
+                        new_job.convertToCReference(None)
+                        ext_job_list.add_job(new_job.name(), new_job)
+                    except:
+                        continue
```
Comment (oysteoh) [inline python/res/enkf/site_config.py:93]: is this exception ok without any logging / printing ? or should we have something which notice us about the job creation is failing..?

File: python/res/enkf/site_config.py [hunk 3]
```diff
@@ -89,13 +123,36 @@ def get_license_root_path(self):
...
+
+            if other_job_list[job_name].get_config_file() != self_job_list[job_name].get_config_file():
+                return False
+
+            if other_job_list[job_name].get_stderr_file() != self_job_list[job_name].get_stderr_file():
+                return False
+
+            if other_job_list[job_name].get_stdout_file() != self_job_list[job_name].get_stdout_file():
+                return False
+        return True
```
Comment (oysteoh) [inline python/res/enkf/site_config.py]: newline

**Review by oysteoh — APPROVED** [review]
looks good to me !

File: python/res/enkf/site_config.py (hunk 2 above)
Comment (ManInFez) [reply python/res/enkf/site_config.py:93]: You are right, should be something to warn the user..

File: python/res/enkf/config_keys.py [hunk 4]
```diff
@@ -123,3 +123,8 @@ class ConfigKeys:
     ECL_FILE         = _ecl_file()
     OUTPUT_FORMAT    = _output_format()
     INIT_FILES       = _init_files()
+    LICENSE_PATH     = "LICENSE_PATH"
```
Comment (markusdregi) [inline python/res/enkf/config_keys.py]: If these are defined in the C-code, please fetch the value from there instead of defining it directly.

**Pull request merged**